src/Libraries/2D/Source/Flat8/fl8hlin.c

```diff
@@ -90,7 +90,7 @@ void gri_flat8_uhline_clut(short x0, short y0, short x1, int32_t c, int32_t parm
         x1 = t;
     }
 
-    c = (int32_t)(((uchar *)parm)[c]);
+    c = (int32_t)(((uchar *)(intptr_t)parm)[c]);
```

Comment:
parm is still being squeezed through a 32-bit value, so this won't solve the problem for 64-bit platforms.

Reply:
Yes, that pretty ugly fix. Still I doubt that flat8 format used by SS, so there should be no any harm.